falcon/testing/test_case.py

```diff
@@ -18,36 +18,79 @@
 utilities for simulating and validating HTTP requests.
 """
 
-try:
-    import testtools as unittest
-except ImportError:  # pragma: nocover
-    import unittest
+import os
+import unittest
+import warnings
 
 import falcon
 import falcon.request
-
-# TODO hoist for backwards compat. Remove in falcon 4.
 from falcon.testing.client import Result  # NOQA
 from falcon.testing.client import TestClient
 
+base_case = os.environ.get('FALCON_BASE_TEST_CASE')
+
+if base_case and 'testtools.TestCase' in base_case:
```

Comment:
We need to actually import the specified module, not just support `testtools.TestCase`.
It can be `something.else.TestCase` too.